### PR TITLE
feat(werklijst): kolommen Afdeling en Klantcontactnummer toevoegen

### LINKED ISSUE #499: 🛠 Task: Kolommen Afdeling en Klantcontactnummer toevoegen
# 🛠 Task: Kolommen "Afdeling" en "Klantcontactnummer" toevoegen

---

## Intent

Als Behandelaar wil ik op "Mijn werkvoorraad" de kolom "Afdeling" zien, zodat ik bij contactverzoeken voor meerdere afdelingen kan onderscheiden voor welke afdeling elk item bedoeld is. Als Coördinator of Behandelaar wil ik het klantcontactnummer (contactmomentnummer) zien, zodat ik een contactverzoek snel kan identificeren en refereren.

---

## Functional Description

Twee kolomuitbreidingen op de werklijsten:

1. **Kolom "Afdeling"** toevoegen aan "Mijn werkvoorraad" (MyInterneTakenTable). Op "Alle contactverzoeken" (AllInterneTakenTable) is deze kolom al aanwezig.
2. **Kolom "Klantcontactnummer"** toevoegen aan beide werklijsten. Dit toont het nummer van het aanleidinggevend klantcontact (`ContactmomentNummer` / `Klantcontact.Nummer`). De data is al beschikbaar in de backend-modellen (`ContactmomentNummer` in `InterneTaakOverzichtItem`, `AanleidinggevendKlantcontact.Nummer` in `MyInterneTaakOverviewItem`) maar wordt niet als kolom weergegeven — het wordt nu alleen gebruikt in de detail-link URL.

---

## Design Reference

N/A — geen aparte Figma. Volgt bestaand NLDS/Utrecht tabelpatroon (`utrecht-table`). Kolommen sluiten aan bij het bestaande stijlpatroon.

---

### In scope

- "Afdeling" kolom toevoegen aan MyInterneTakenTable
- Backend: `AfdelingNaam` beschikbaar maken in `MyInterneTaakOverviewItem` (nu ontbreekt dit veld)
- Backend: `MyInterneTakenOverviewService` uitbreiden om afdeling-info te mappen
- "Klantcontactnummer" kolom toevoegen aan AllInterneTakenTable en MyInterneTakenTable
- Kolom toont `ContactmomentNummer` (het nummer van het aanleidinggevend klantcontact)

### Out of scope

- Sorteren op de nieuwe kolommen (→ apart task "Sorteren op kolomhoofden")
- Wijzigingen aan "Alle contactverzoeken" voor de Afdeling-kolom (deze bestaat al)

---

### Request

Geen wijziging — bestaande endpoints.

### Response

Uitbreiding van `MyInterneTaakOverviewItem` response met `afdelingNaam`:

```json
{
  "afdelingNaam": "Burgerzaken",
  "..."
}
```

`ContactmomentNummer` is al beschikbaar:
- In `InterneTaakOverzichtItem` als `contactmomentNummer`
- In `MyInterneTaakOverviewItem` via `aanleidinggevendKlantcontact.nummer`

### Events / Side Effects

Geen.

---

## Technical Approach

- Architecture decisions: `MyInterneTaakOverviewItem` uitbreiden met `AfdelingNaam`. In `MyInterneTakenOverviewService` de afdeling-informatie afleiden uit de actor-gegevens van de internetaak, vergelijkbaar met hoe `InterneTakenOverzichtService` dit al doet.
- Data flow: `Internetaak.ToegewezenAanActoren` → actortype-filtering → `AfdelingNaam` → `MyInterneTaakOverviewItem` → Vue kolom
- Trade-offs: We hergebruiken het bestaande patroon uit `InterneTakenOverzichtService` voor het ophalen van afdeling-info. Dit houdt de logica consistent.

### Aggregates & Invariants

- Aggregates: Geen
- Invariants: Geen

---

## Impacted Areas

- Backend: `MyInterneTaakOverviewModel.cs` (nieuw veld `AfdelingNaam`), `MyInterneTakenOverviewService.cs` (afdeling-mapping)
- Frontend: `MyInterneTakenTable.vue` (kolommen Afdeling en Klantcontactnummer toevoegen), `AllInterneTakenTable.vue` (kolom Klantcontactnummer toevoegen)
- Database: Geen
- APIs: `GET /api/internetaken/aan-mij-toegewezen` (response-uitbreiding met `afdelingNaam`)
- External systems: Geen

---

## Gherkin

```gherkin
Feature: Kolommen Afdeling en Klantcontactnummer op werklijsten

  Scenario: Behandelaar ziet kolom Afdeling op Mijn werkvoorraad
    Given een Behandelaar heeft contactverzoeken voor meerdere afdelingen
    When de Behandelaar "Mijn werkvoorraad" bekijkt
    Then toont de werklijst een kolom "Afdeling"
    And toont elk contactverzoek de afdelingsnaam waarvoor het bedoeld is

  Scenario: Klantcontactnummer zichtbaar op Alle contactverzoeken
    Given een Coördinator bekijkt "Alle contactverzoeken"
    When de werklijst contactverzoeken toont
    Then bevat de lijst een kolom met het contactmomentnummer
    And toont elk contactverzoek het nummer van het aanleidinggevend klantcontact

  Scenario: Klantcontactnummer zichtbaar op Mijn werkvoorraad
    Given een Behandelaar bekijkt "Mijn werkvoorraad"
    When de werklijst contactverzoeken toont
    Then bevat de lijst een kolom met het contactmomentnummer
    And toont elk contactverzoek het nummer van het aanleidinggevend klantcontact
```

---

## Edge Cases & Risks

- Contactverzoek zonder gekoppelde afdeling — toon lege cel, geen foutmelding
- Klantcontact zonder nummer — toon lege cel (zou niet moeten voorkomen)

### Edge Case Results

| Edge Case | Result | Bug Filed |
| --------- | ------ | --------- |
| Contactverzoek zonder gekoppelde afdeling | ✅ | — |
| Klantcontact zonder nummer | ✅ | — |

---

## Observability

- Logs: None required for this task
- Metrics: None required for this task
- Alerts: None required for this task

---

## Dependencies

Geen harde afhankelijkheid. Kan parallel aan of na Task #498 (urgentielabel) worden geïmplementeerd.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: Additieve response-uitbreiding, niet breaking
- Data migration: N/A

---

## Definition of Done

- [x] Linting passes (`npm run lint:ci`)
- [ ] Formatting passes (`npm run format:ci`)
- [x] All Gherkin scenarios correctly implemented (code inspection)
- [x] Edge cases handled
- [ ] PR approved by at least one reviewer
- [ ] No unrelated changes or scope creep

---

## Test Mapping

Phase 1 (code inspection, this QA pass) — E2E automation is Phase 2, tracked in #509.

| Scenario | Result | Bug Filed |
| -------- | ------ | --------- |
| Behandelaar ziet kolom Afdeling op Mijn werkvoorraad | ✅ | — |
| Klantcontactnummer zichtbaar op Alle contactverzoeken | ✅ | — |
| Klantcontactnummer zichtbaar op Mijn werkvoorraad | ✅ | — |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

Behandelaars die contactverzoeken voor meerdere afdelingen verwerken hebben zicht nodig op de afdelingsnaam per item. Daarnaast willen zowel Coördinatoren als Behandelaars het klantcontactnummer (contactmomentnummer) zien om contactverzoeken snel te identificeren en refereren. Dit is onderdeel van Feature #402 (werklijst urgentie, sorteren & filteren).

## Changes

- **Backend model uitbreiding**: `AfdelingNaam` veld toegevoegd aan `MyInterneTaakOverviewModel`
- **Backend service uitbreiding**: `MyInterneTakenOverviewService` leidt afdeling-informatie af uit de actor-gegevens van de internetaak (zelfde patroon als `InterneTakenOverzichtService`)
- **Frontend — Mijn werkvoorraad**: kolommen "Afdeling" en "Klantcontactnummer" toegevoegd aan `MyInterneTakenTable.vue`
- **Frontend — Alle contactverzoeken**: kolom "Klantcontactnummer" toegevoegd aan `AllInterneTakenTable.vue`

## Test plan

Per [QA.md Phase 1](docs/steering/QA.md#phase-1-implementation-verification-feature-branch-pre-merge) — verificatie via code inspection. E2E tests volgen in Phase 2 na deploy.

- [ ] Behandelaar ziet kolom Afdeling op Mijn werkvoorraad
- [ ] Klantcontactnummer zichtbaar op Alle contactverzoeken
- [ ] Klantcontactnummer zichtbaar op Mijn werkvoorraad
- [ ] Edge case: contactverzoek zonder afdeling toont lege cel
- [ ] Edge case: klantcontact zonder nummer toont lege cel

## Related

Closes #499

### CODE REVIEW

File: InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/MyInterneTakenTable.vue [hunk 1]
```diff
@@ -28,6 +30,12 @@
         <utrecht-table-cell>
           <urgentie-badge :urgentie="taak.urgentie" />
         </utrecht-table-cell>
+        <utrecht-table-cell class="text-truncate" :title="taak.afdelingNaam || ''">
+          {{ taak.afdelingNaam || "-" }}
+        </utrecht-table-cell>
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/MyInterneTakenTable.vue:35]: Addressed in 3462d99 — added `afdelingNaam?: string | null` to `MyInterneTaakOverviewItem` in `src/types/internetaken.ts`.

File: InterneTaakAfhandeling.Web.Server/Features/MyInterneTakenOverview/MyInterneTakenOverviewService.cs [hunk 2]
```diff
@@ -56,10 +61,42 @@ private MyInterneTaakOverviewItem MapToOverviewItem(Internetaak internetaak)
...
+        {
+            try
+            {
+                return await _openKlantApiClient.GetActorAsync(uuid);
+            }
+            catch
+            {
+                return null;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Server/Features/MyInterneTakenOverview/MyInterneTakenOverviewService.cs:101]: Addressed in 3462d99 — added `ILogger<MyInterneTakenOverviewService>` and now catching `Exception ex` with `_logger.LogWarning(ex, ...)`, aligned with `InterneTakenOverzichtService.GetActorAsync` pattern.
